king_phisher/catalog.py

```diff
@@ -511,6 +511,22 @@ def add_catalog_url(self, url):
 			return
 		return catalog
 
+	def add_catalog_dict(self, dict_):
```

Comment:
On second thought, this method and the `add_catalog_url` methods should probably both be removed in favor of a new method like:

```
def add_catalog(catalog):
    self.catalogs[catalog.id] = catalog
```

This would then require the callers to implement the error handling so this doesn't silently fail when used by others.